src/views/Home/components/DaysLeft/index.jsx

```diff
@@ -24,65 +33,145 @@ const DaysLeft = () => {
       ]);
       const first = format(parseISO(currentSession.SessionBeginDate), 'MM/dd/yyyy');
       const last = format(parseISO(currentSession.SessionEndDate), 'MM/dd/yyyy');
-      const currentSessionDescription = currentSession.SessionDescription.replace(
-        /(Academic Year)|(Grad)/gm,
-        '',
-      );
+      const fallBegin = new Date((await session.get(`${currentYear - 1}09`)).SessionBeginDate);
+      const fallEnd = new Date((await session.get(`${currentYear - 1}09`)).SessionEndDate);
+      const springBegin = new Date((await session.get(`${currentYear}01`)).SessionBeginDate);
+      const springEnd = new Date((await session.get(`${currentYear}01`)).SessionEndDate);
+      const summerBegin = new Date((await session.get(`${currentYear}05`)).SessionBeginDate);
+      const summerEnd = new Date((await session.get(`${currentYear}05`)).SessionEndDate);
+      const nextFallBegin = new Date((await session.get(`${currentYear}09`)).SessionBeginDate);
 
```

Comment:
```suggestion
      const fallBegin = new Date((await session.get(`${currentYear - 1}09`)).SessionBeginDate);
      const fallEnd = new Date((await session.get(`${currentYear - 1}09`)).SessionEndDate);
      const springBegin = new Date((await session.get(`${currentYear}01`)).SessionBeginDate);
      const springEnd = new Date((await session.get(`${currentYear}01`)).SessionEndDate);
      const summerBegin = new Date((await session.get(`${currentYear}05`)).SessionBeginDate);
      const summerEnd = new Date((await session.get(`${currentYear}05`)).SessionEndDate);
      const nextFallBegin = new Date((await session.get(`${currentYear}09`)).SessionBeginDate);

```
This seems wildly inefficient. Also the magic numbers are a bit of a concern. You could potentially store all beginning/end dates in a single JSON object with a useMemo hook to prevent outstanding memory usage. (there are plenty of tutorials on how to use [useMemo](https://www.freecodecamp.org/news/how-to-work-with-usememo-in-react/))